src/mrack/providers/provider.py

```diff
@@ -432,7 +436,7 @@ async def strategy_retry(self, reqs):
         success_hosts = []
         error_hosts = []
         delta_sleep = global_context.CONFIG.delta_sleep
-        max_utilization = global_context.CONFIG.max_utilization
+        max_utilization = self._get_max_utilization()
```

Comment:
One nitpick, could you please extract this to another 'refactor' commit please?

Reply:
No problem, updated